### DISCUSSION TITLE
Multiline text editing

### DISCUSSION THREAD

**Discussion by medecau** [Textualize/rich#1030] [Ideas] [answered]
Hi 👋,

*I have searched the issues and discussions for some keywords that might be associated with this and could not find anything.*

Anyway, was wondering if it is currently possible to create something like the `nano` text editor using `rich`. I guess it'd require something similar to a `pager` with an `insert` cursor and a `rope` data structure.

*I'll admit it sounds a bit outside the goals of `rich`.* Am I expecting too much?

**Comment by willmcgugan** [comment, marked as answer]
Not currently. Although I do planning on adding more dynamic features, and an editor would be on the top of the list...

BTW Thanks for the sponsor!

**Reply by medecau** [reply]
To clarify the feature-request side of my message above. The text-editor I'd like to use would be more user-friendly than advanced-capable – think `nano` instead of `vim` or `emacs`.
